smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSource.java

```diff
@@ -59,14 +68,47 @@ public KafkaSource(Vertx vertx, KafkaConnectorIncomingConfiguration config) {
         kafkaConfiguration.remove("broadcast");
         kafkaConfiguration.remove("partitions");
 
-        this.consumer = KafkaConsumer.create(vertx, kafkaConfiguration);
+        final KafkaConsumer<K, V> kafkaConsumer = KafkaConsumer.create(vertx, kafkaConfiguration);
+        final List<Publisher<KafkaConsumerRecord<K, V>>> multis = new ArrayList<>();
+        config
+                .getConsumerRebalanceListenerName()
+                .map(name -> {
+                    log.info("Loading KafkaConsumerRebalanceListener " + name);
+                    return NamedLiteral.of(name);
+                })
+                .map(consumerRebalanceListeners::select)
+                .map(Instance::get)
+                .ifPresent(listener -> multis.add(Multi
```

Comment:
Do we want to handle the case with multiple listeners for a single topic?

Reply:
There won't be multiple listeners. But the listener is called whenever there's a rebalance not just during startup.

Reply:
Got it!